tools/cargo-miden/src/target.rs

```diff
@@ -6,6 +6,62 @@ use std::{
 
 use anyhow::{bail, Result};
 
+/// Represents whether the Cargo project is a Miden program or a library.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ProjectType {
+    Program,
+    Library,
+}
+
+/// Represents different types of rollup components that can be targeted
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum RollupTarget {
+    /// An account contract in the Miden rollup
+    Account,
+    /// A note script in the Miden rollup
+    NoteScript,
+}
+
+/// Represents the execution environment for the compiled code
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum TargetEnvironment {
```

Comment:
This is basically a duplicate of `TargetEnv`, any reason not to consolidate them?

Reply:
Oh, I totally missed that. Removed mine in favor of `TargetEnv`.